### ISSUE TITLE
EventBridgeRule RetryPolicy for Targets

### ISSUE THREAD

**Issue by jeffmarcinko** [aws/serverless-application-model#1857]
**Request to Add RetryPolicy to the EventBridgeRule property:**

Please add support for EventBridge retry policy to the EventBridgeRule sub-property of the Serverless Function resource.  This should map to the CloudFormation Target -> RetryPolicy property (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-events-rule-target.html#cfn-events-rule-target-retrypolicy)

Desired Example:

```yaml
  EventProcessingFunction:
    Type: AWS::Serverless::Function 
    Properties:
      CodeUri: src/event_processing
      Handler: app.handler
      Runtime: python3.8
      Events:
        NewEventCreatedRule:
          Type: EventBridgeRule
          Properties:
            EventBusName: !Ref ProcessingEventBus
            Pattern:
              source:
                - com.example.service
              detail-type:
                - NewExampleCreated
            DeadLetterConfig:  # Event Delivery DLQ
              Arn: !GetAtt BuildEventBusDLQ.Arn
            RetryPolicy:  # Event Delivery Retry Policy
              MaximumEventAgeInSeconds: 60 
              MaximumRetryAttempts: 2
```

**Comment by mndeveci** [comment]
Thanks for raising this request @jeffmarcinko ,

RetryPolicy have been added to `EventBridgeRule` event source types recently, it can be found in this PR; https://github.com/aws/serverless-application-model/pull/1842

So feature will be released with the next scheduled release of the aws-sam-translator.

**Comment by mgrandis** [comment]
Released in 1.34.0:
https://github.com/aws/serverless-application-model/releases/tag/v1.34.0
